src/lib.rs

```diff
@@ -92,32 +95,32 @@ fn get_compact_symbol_table_impl(
     binary_data: &[u8],
     debug_data: &[u8],
     breakpad_id: &str,
-) -> Option<compact_symbol_table::CompactSymbolTable> {
+) -> Result<compact_symbol_table::CompactSymbolTable> {
     let mut reader = Cursor::new(binary_data);
-    if let Ok(hint) = goblin::peek(&mut reader) {
-        match hint {
-            Hint::Elf(_) => {
-                return elf::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::Mach(_) => {
-                return macho::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::MachFat(_) => {
-                let multi_arch = mach::MultiArch::new(binary_data).ok()?;
-                for fat_arch in multi_arch.iter_arches().filter_map(Result::ok) {
-                    let arch_slice = fat_arch.slice(binary_data);
-                    if let Some(table) = macho::get_compact_symbol_table(arch_slice, breakpad_id) {
-                        return Some(table);
-                    }
+    match goblin::peek(&mut reader)? {
+        Hint::Elf(_) => elf::get_compact_symbol_table(binary_data, breakpad_id),
+        Hint::Mach(_) => macho::get_compact_symbol_table(binary_data, breakpad_id),
+        Hint::MachFat(_) => {
+            let multi_arch = mach::MultiArch::new(binary_data).ok();
+            for fat_arch in multi_arch
+                .unwrap()
```

Comment:
You added an `.unwrap()` here - can you propagate the error instead?

Reply:
We'd bump into the error we discussed earlier-- `NoneError is not implemented ...`. I think we'd need to have option here...

Reply:
Oh, that's because you're calling it on an `Option`, which you obtained by calling `.ok()` in the line above. Right?
What happens if you remove both the unwrap and the ok, and put the question mark where the `.ok()` was?